### PR TITLE
Bug Fixes for Xeno Tunnels

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Fixes xeno tunnel bugs related to keeping the dialogue open, as well as proofing tunnels against future possible bugs/unintended behaviours.

## Why It's Good For The Game

Fixes xeno tunnel bugs related to keeping the dialogue open, as well as proofing tunnels against future possible bugs/unintended behaviours.

Fixes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/5721

## Changelog
:cl:
fix: Fixes a bug related to keeping the xeno tunnel selection interface open.
code: Proofs xeno tunnels against future bugs/unintended behaviours.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/xeno.dm [hunk 1]
```diff
@@ -645,6 +645,10 @@ TUNNEL
 	hud_set_xeno_tunnel()
 
 /obj/structure/tunnel/Destroy()
+	var/drop_loc = get_turf(src)
+	for(var/atom/movable/thing as() in src) //Empty the tunnel of contents
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:649]:
```suggestion
	for(var/atom/movable/thing as() in contents) //Empty the tunnel of contents
```

File: code/game/objects/structures/xeno.dm [hunk 2]
```diff
@@ -750,9 +761,13 @@ TUNNEL
 		if(targettunnel && isturf(targettunnel.loc)) //Make sure the end tunnel is still there
 			M.forceMove(targettunnel)
 			var/double_check = input(M, "Emerge here?", "Tunnel: [targettunnel]") as null|anything in list("Yes","Pick another tunnel")
+			if(!targettunnel) //double check that the tunnel we're now in exists in the event it gets destroyed while we still have the interface open
+				return
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:765]: This needs to be qdeleted

File: code/game/objects/structures/xeno.dm [hunk 3]
```diff
@@ -750,9 +761,13 @@ TUNNEL
...
+			if(M.loc != targettunnel) //double check that we're still in the tunnel in the event it gets destroyed while we still have the interface open
+				return
 			if(double_check == "Pick another tunnel")
 				return targettunnel.pick_a_tunnel(M)
-			else //Whether we say yes or cancel out of it
+			else //Whether we say yes or cancel out of it;
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:770]: ??

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/xeno.dm (hunk 3 above)
Comment (Surrealaser) [reply code/game/objects/structures/xeno.dm:770]: I think there was additional commentary I added that was later deleted

**Review by SplinterGP — APPROVED** [review]

File: code/game/objects/structures/xeno.dm [hunk 4]
```diff
@@ -720,7 +724,14 @@ TUNNEL
 ///Here we pick a tunnel to go to, then travel to that tunnel and peep out, confirming whether or not we want to emerge or go to another tunnel.
 /obj/structure/tunnel/proc/pick_a_tunnel(mob/living/carbon/xenomorph/M)
 	var/obj/structure/tunnel/targettunnel = input(M, "Choose a tunnel to crawl to", "Tunnel") as null|anything in GLOB.xeno_tunnels
-	if(!targettunnel)
+	if(!src) //Make sure we still exist in the event the player keeps the interface open
+		return
+	if(!M.Adjacent(src) && M.loc != src) //Make sure we're close enough to our tunnel; either adjacent to or in one
+		return
+	if(!targettunnel) //Make sure our target destination still exists in the event the player keeps the interface open
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:731]: this is still not fixed

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by SplinterGP — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is now above zero (1). Feel free to make Feature/Tweak/Balance PRs.

**Pull request merged**